### PR TITLE
feat: Introduce isBooleanable function.

### LINKED ISSUE #340: Feature Request: isBooleanable()
I would love to see a function that tells you if an object/string/whatever is one of the "booleanable" options.

For example isBooleanable() would return True for the values you consider a truth or false: true, false, 'Y', 'N', '1', 0, 'on', 'off', etc...

But when passing something that isn't one of the covered true/false items, it would return False: 'computer', '2022-02-15T11:37:00-08:00', etc...

That would allow you to check if an object is something that should evaluate to a true/false before calling boolean() and potentially getting False as the default result.

### PR DESCRIPTION
**Pull request by goloroden**
This PR introduces a new function (`isBooleanable`) which checks whether a given value can be considered to be a boolean.

Closes #340

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,10 +56,36 @@ The `boolean` function considers the following values to be equivalent to `true`
...
 In addition to the primitive types mentioned above, boolean also supports their object wrappers `Boolean`, `String`, and `Number`.
 
-_Please note that if you provide a `string` or `String` object, it will be trimmed._
+_Please note that if you provide a `string` or a `String` object, it will be trimmed._
 
 All other values, including `undefined` and `null` are considered to be `false`.
 
+### Figuring out whether a value can be considered to be boolean
+
+From time to time, you may not want to directly convert a value to its boolean equivalent, but explicitly check whether it looks like a boolean. E.g., although `boolean('F')` return `false`, the string `F` at least looks like a boolean, in contrast to something such as `123` (for which `boolean(123)` would also return `false`).
```
Comment (strangedev) [inline README.md:65]: Small typo here.
> although `boolean('F')` return `false`

I think it should probably read:
> although `boolean('F')` returns `false`

File: test/unit/isBooleanable.ts [hunk 2]
```diff
@@ -0,0 +1,288 @@
+import { assert } from 'assertthat';
```
Comment (strangedev) [inline test/unit/isBooleanable.ts:1]: Your test file is missing the `Tests` suffix and the tests did not execute. Can you please rename the file? :slightly_smiling_face:

**Review by strangedev — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (goloroden) [reply README.md]: Good catch 😊

**Comment by goloroden** [comment]
@strangedev Thanks for your feedback! I adjusted the code according to your feedback. Could you please have another look?

File: test/unit/isBooleanableTests.ts [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    test('null returns false.', async (): Promise<void> => {
+      assert.that(isBooleanable(null)).is.false();
+    });
+
+    test('{} returns false.', async (): Promise<void> => {
+      assert.that(isBooleanable({})).is.false();
+    });
+  });
+
+  suite('isBooleanable', (): void => {
```
Comment (strangedev) [inline test/unit/isBooleanableTests.ts:22]: I think this suite is not named correctly. Shouldn't this be `boolean` instead of `isBooleanable`?

**Review by strangedev — CHANGES_REQUESTED** [review]
Just one small thing :slightly_smiling_face:

File: test/unit/isBooleanableTests.ts (hunk 3 above)
Comment (goloroden) [reply test/unit/isBooleanableTests.ts]: Yes, of course 🙈

Fixed it.

**Review by strangedev — APPROVED** [review]

**Pull request merged**